160801263 Implement user signup, sign in and sign out on api

AwaMelvine opened the pull request:
#### What does this PR do?

Implements user authentication, sign up, sign in and sign out on the API.

#### Description of Task to be completed?
Create API endpoints to
- Sign up User
- Sign in User
- logout user
- Validate user

#### How should this be manually tested?
Using postman, test the following API endpoints as follows:

1. POST - /api/v1/auth/signup  -- For user sign up
Request body should have the following data
```
{
   role:'user',
   username:'Jane',
   email:'testmail@gmail.com',
   password:'john'
   passwordConf:'john'
}
```
If signup was successful, token will be returned orthwise a signin fail message.

2.  POST - /api/v1/auth/signin -- This signs user in
Request body should have the following data
```
{
   email:'testmail@gmail.com',
   password:'john'
}
```
If credentials are correct, a token will be returned else a login fail message

3. GET - `/api/v1/logout`  -- This logs user out

#### What are the relevant pivotal tracker stories?
[#160801263] User can manage accounts, signup, sign in and sign out

robocopkaka commented on api/v1/models/User.js:
```diff
@@ -0,0 +1,98 @@
+      const { rows } = await db.query(queryString, params);
+      return rows;
+    } catch (error) {
+      console.log(error);
```
Avoid leaving `console.logs` in your code when you push to github

AwaMelvine replied:
Okay. Noted!

AwaMelvine committed:
[#160801263] Implement user sign up and login

- Set up User model and write query methods
- create signup and login endpoints
- return a json web token when user signs up or logs in

[#160801263]

AwaMelvine committed:
[#160801263] Implement input validation on auth endpoints

Write validation middlewares for signup and sign in endpoints

[#160801263]

AwaMelvine committed:
[#160801263] Implement user sign up, sign in and sign out

Use jsonwebtokens to implement authentication.

[#160801263]

AwaMelvine committed:
[#160801263] Require authentication on place order endpoint

A user must be logged in before they can place an order for food.

[#160801263]

AwaMelvine committed:
[#160815583] Fix issue with tests

Configure babel for support of the necessary es6 features

[Fixes#160815583]
